cmd/k8s-operator/connector.go

```diff
@@ -0,0 +1,262 @@
+		if err := a.Update(ctx, cn); err != nil {
+			return reconcile.Result{}, err
+		}
+		logger.Infof("connector resources cleaned up")
+		return reconcile.Result{}, nil
+	}
+
+	oldCnStatus := cn.Status.DeepCopy()
+	defer func() {
+		if cn.Status.SubnetRouter == nil {
+			tsoperator.SetConnectorCondition(cn, tsapi.ConnectorReady, metav1.ConditionUnknown, "", "", cn.Generation, a.clock, logger)
+		} else if cn.Status.SubnetRouter.Ready == metav1.ConditionTrue {
+			tsoperator.SetConnectorCondition(cn, tsapi.ConnectorReady, metav1.ConditionTrue, reasonSubnetRouterCreated, reasonSubnetRouterCreated, cn.Generation, a.clock, logger)
+		} else {
+			tsoperator.SetConnectorCondition(cn, tsapi.ConnectorReady, metav1.ConditionFalse, cn.Status.SubnetRouter.Reason, cn.Status.SubnetRouter.Reason, cn.Generation, a.clock, logger)
+		}
+		if !apiequality.Semantic.DeepEqual(oldCnStatus, cn.Status) {
+			// an error encountered here should get returned by the Reconcile function
+			if updateErr := a.Client.Status().Update(ctx, cn); updateErr != nil {
+				err = updateErr
+			}
+		}
+	}()
+
+	if !slices.Contains(cn.Finalizers, FinalizerName) {
+		// This log line is printed exactly once during initial provisioning,
+		// because once the finalizer is in place this block gets skipped. So,
+		// this is a nice place to tell the operator that the high level,
+		// multi-reconcile operation is underway.
+		logger.Infof("ensuring connector is set up")
+		cn.Finalizers = append(cn.Finalizers, FinalizerName)
+		if err := a.Update(ctx, cn); err != nil {
+			err = fmt.Errorf("failed to add finalizer: %w", err)
+			logger.Errorf("error adding finalizer: %v", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// At the momement SubnetRouter is the only component configurable via
+	// ConnectorSpec and a ConnectorSpec without a SubnetRouter is invalid- but
+	// that will change in the future - so run the cleanup and provision
+	// conditionally.
+	if cn.Spec.SubnetRouter != nil && len(cn.Spec.SubnetRouter.Routes) > 0 {
+		var sb strings.Builder
+		sb.WriteString(string(cn.Spec.SubnetRouter.Routes[0]))
+		for _, r := range cn.Spec.SubnetRouter.Routes[1:] {
+			sb.WriteString(fmt.Sprintf(",%s", r))
+		}
+		cidrsS := sb.String()
+		logger.Debugf("ensuring a subnet router is deployed")
+		err := a.maybeProvisionSubnetRouter(ctx, logger, cn, cidrsS)
+		if err != nil {
+			msg := fmt.Sprintf(messageSubnetRouterCreationFailed, cidrsS, err)
+			cn.Status.SubnetRouter = &tsapi.SubnetRouterStatus{
+				Ready:   metav1.ConditionFalse,
+				Reason:  reasonSubnetRouterCreationFailed,
+				Message: msg,
+			}
+			a.recorder.Eventf(cn, corev1.EventTypeWarning, reasonSubnetRouterCreationFailed, msg)
+			return reconcile.Result{}, err
+		} else {
+			cn.Status.SubnetRouter = &tsapi.SubnetRouterStatus{
+				Routes:  cidrsS,
+				Ready:   metav1.ConditionTrue,
+				Reason:  reasonSubnetRouterCreated,
+				Message: fmt.Sprintf(messageSubnetRouterCreated, cidrsS),
+			}
+		}
+
```

Comment:
if you return here then you can unindent the block below too.

Reply:
I will set subnetrouter as required and remove the block below altogether. When we start supporting more component types, we can anyway _not_ support removing one component definition from a Connector and adding another.